src/utils.c

```diff
@@ -2,99 +2,37 @@
 #include "cx.h"
 #include <stdbool.h>
 #include <stdlib.h>
+
+#include "lib_standard_app/crypto_helpers.h"
+#include "ledger_assert.h"
+
 #include "utils.h"
 
-void get_public_key(uint8_t *publicKeyArray, const uint32_t *derivationPath, size_t pathLength) {
-    cx_ecfp_private_key_t privateKey;
-    cx_ecfp_public_key_t publicKey;
+void get_public_key(uint8_t publicKeyArray[static PUBKEY_LENGTH],
+                    const uint32_t *derivationPath,
+                    size_t pathLength) {
+    uint8_t rawPubkey[65];
 
-    get_private_key(&privateKey, derivationPath, pathLength);
-    BEGIN_TRY {
-        TRY {
-            cx_ecfp_generate_pair(CX_CURVE_Ed25519, &publicKey, &privateKey, 1);
-        }
-        CATCH_OTHER(e) {
-            MEMCLEAR(privateKey);
-            THROW(e);
-        }
-        FINALLY {
-            MEMCLEAR(privateKey);
-        }
+    if (CX_OK != bip32_derive_with_seed_get_pubkey_256(HDW_ED25519_SLIP10,
+                                                       CX_CURVE_Ed25519,
+                                                       derivationPath,
+                                                       pathLength,
+                                                       rawPubkey,
+                                                       NULL,
+                                                       CX_SHA512,
+                                                       NULL,
+                                                       0)) {
+        LEDGER_ASSERT(false, "Fail to get pub key");
```

Comment:
Do we really want to use assert for this kind of stuff ? It could make sense to send back an error code here